packages/components/src/custom-select-control/test/index.tsx

```diff
@@ -689,3 +689,128 @@ describe.each( [
 		} );
 	} );
 } );
+
+describe( 'Type checking', () => {
```

Comment:
Not something for this PR, but I wonder if we should start putting these type-checking pseudo-tests in a separate file under `/test` (types.tsx for example).

Then we could add a global ESLint disable rule for `jest/expect-expect` and we'll no longer have to do it every time. 

Thoughts?

Reply:
cc @mirka and @DaniGuardiola

Reply:
Extracted to #64588